lib/internal/url.js

```diff
@@ -309,8 +310,10 @@ class URL {
   }
 
   set search(search) {
+    search = String(search);
+    if (search[0] === '?') search = search.slice(1);
     update(this, search);
-    this[searchParams][searchParams] = querystring.parse(this.search);
+    this[searchParams][searchParams] = querystring.parse(this[context].query);
```

Comment:
This line is more for consistency with the constructor and cosmetics than anything else: it's a bit odd to see the property being read in its setter.